app/scripts/lib/typed-message-manager.js

```diff
@@ -209,6 +218,13 @@ export default class TypedMessageManager extends EventEmitter {
             `Provided chainId "${chainId}" must match the active chainId "${activeChainId}"`,
           );
         }
+        const { domain } = data;
+        assert.equal(
+          domain?.name,
+          new URL(params?.origin).hostname,
```

Comment:
`new URL` can throw if it's passed bad information, so I'm thinking we need a `try/catch` here or simply do `params.origin`.  Do we know if this info could be absent?

Reply:
At the moment at least it's guaranteed to be present. It's set by the browser when we receive a message.

We might have to rethink this if/when we start handling messages from snaps though. Unless we give them valid origins too.